### DISCUSSION TITLE
object has no attribute 'id' when trying to db.session.merge

### DISCUSSION THREAD

**Discussion by dremdem** [sqlalchemy/sqlalchemy#6857] [Usage Questions] [answered]
### Describe the bug

When I tried to merge and flush one object it raises: 

```
'GoogleTask' object has no attribute 'id'
```

### To Reproduce

```python
class GoogleTask(db.Model, TimestampMixin):
    __tablename__ = 'google_task'

    uuid = db.Column(db.String(36), nullable=False, primary_key=True)
    complete = db.Column(db.Boolean, default=False, nullable=False)
    title = db.Column(db.String(500), nullable=False)
    note = db.Column(db.Text())
    created_at = db.Column(db.DateTime, nullable=False)
    completed_at = db.Column(db.DateTime)
    practice_id = db.Column(db.Integer, db.ForeignKey('practices.id'))
    practice = db.relationship('Practices')
    last_name = db.Column(db.String(50))
    first_name = db.Column(db.String(50))
    birthday = db.Column(db.Date)
    service_date = db.Column(db.DateTime)
    biller_done = db.Column(db.Boolean, nullable=False)
    submitted_amount = db.Column(db.DECIMAL(10, 2))


...
task = dict(<dict with the right values>)

db.session.merge(GoogleTask(**task))
db.session.flush()

...
```

### Error

'GoogleTask' object has no attribute 'id'

### Versions

- OS: Macos Catalina 10.15.7
- Python: 3.7.2
- SQLAlchemy: 1.3.22
- Database: MySQL
- DBAPI:

**Comment by zzzeek** [comment, marked as answer]
hi i converted to discussion as there is no detail provided here to create a bug report from this.

There's no name "id" mentioned in the code you given so this is not an error that SQLAlchemy would raise unless somewhere in your app this attribute is being referred towards.  

therefore if you can provide adequate detail, such as complete stack trace, full code that reproduces, we can possibly determine the cause of this error.

**Reply by dremdem** [reply]
Hi! 

Just founded that you're right: somewhere in the code, there is the additional property "id" that should be defined for every model. 

I'm sorry for my misunderstanding.

Cheer,
Vlad.
